File: LocationVoiture/src/main/java/model/Location.java
package model;

import java.time.LocalDate;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
public class Location {

	@JsonView(JsonViews.Common.class)
	@Id 
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	
	@JsonView(JsonViews.Common.class)
	@Column(nullable = false)
	private LocalDate dateDebut;
	
	@JsonView(JsonViews.Common.class)
	@Column(nullable = false)
	private LocalDate dateFin;
	
	@JsonView(JsonViews.Common.class)
	@Column(columnDefinition = "DECIMAL(7,2)",nullable = false)
	private double prixTotal;

	@JsonView(JsonViews.Common.class)
	@OneToOne
	private Annonce annonce;
	
	@JsonView(JsonViews.Common.class)
	@OneToOne
	private Client client;
	
	
	
	public Location() {
	}


	public Location(LocalDate dateDebut, LocalDate dateFin, double prixTotal) {
		this.dateDebut = dateDebut;
		this.dateFin = dateFin;
		this.prixTotal = prixTotal;
	}


	public Integer getId() {
		return id;
	}


	public void setId(Integer id) {
		this.id = id;
	}


	public LocalDate getDateDebut() {
		return dateDebut;
	}


	public void setDateDebut(LocalDate dateDebut) {
		this.dateDebut = dateDebut;
	}


	public LocalDate getDateFin() {
		return dateFin;
	}


	public void setDateFin(LocalDate dateFin) {
		this.dateFin = dateFin;
	}


	public double getPrixTotal() {
		return prixTotal;
	}


	public void setPrixTotal(double prixTotal) {
		this.prixTotal = prixTotal;
	}
	

	public Annonce getAnnonce() {
		return annonce;
	}


	public void setAnnonce(Annonce annonce) {
		this.annonce = annonce;
	}


	public Client getClient() {
		return client;
	}


	public void setClient(Client client) {
		this.client = client;
	}


	@Override
	public String toString() {
		return "Location [id=" + id + ", dateDebut=" + dateDebut + ", dateFin=" + dateFin + ", prixTotal=" + prixTotal
				+ "]";
	}
	
	
	
	
}

File: LocationVoitureAngular/src/app/location/model/location.ts
import { Annonce } from './annonce';
import { Client } from './client';

export class Location {
  public get client(): Client | undefined {
    return this._client;
  }
  public set client(value: Client | undefined) {
    this._client = value;
  }
  public get annonce(): Annonce | undefined {
    return this._annonce;
  }
  public set annonce(value: Annonce | undefined) {
    this._annonce = value;
  }
  public get assurance(): string | undefined {
    return this._assurance;
  }
  public set assurance(value: string | undefined) {
    this._assurance = value;
  }
  public get prixTotal(): number | undefined {
    return this._prixTotal;
  }
  public set prixTotal(value: number | undefined) {
    this._prixTotal = value;
  }
  public get dateFin(): Date | undefined {
    return this._dateFin;
  }
  public set dateFin(value: Date | undefined) {
    this._dateFin = value;
  }
  public get dateDebut(): Date | undefined {
    return this._dateDebut;
  }
  public set dateDebut(value: Date | undefined) {
    this._dateDebut = value;
  }
  public get id(): number | undefined {
    return this._id;
  }
  public set id(value: number | undefined) {
    this._id = value;
  }

  constructor(
    private _id?: number,
    private _dateDebut?: Date,
    private _dateFin?: Date,
    private _prixTotal?: number,
    private _assurance?: string,
    private _annonce?: Annonce,
    private _client?: Client
  ) {}
}
